File: python/xmppelement.py
class XmppElement(object):
    def __init__(self):
        self.tag = None
        self.attributes = []
        self.text = None
        self.children = []

    def find_child_with_tag(self, tag):
        # type: (str) -> XmppElement
        """

        :rtype: XmppElement
        """
        for c in self.children:
            if c.tag == tag:
                return c
        return None

    def toXml(self, indent=0):
        spaces = " " * indent
        attributes = ""
        for a in self.attributes.getNames():
            attributes += "{0}={1} ".format(a, self.attributes.getValue(a))
        result = spaces + "<{0} {1}".format(self.tag, attributes).strip()
        if self.children or self.text:
            result += ">\n"
            for child in self.children:
                result += child.toXml(indent=indent + 4)
            if self.text:
                result += " " * (indent + 4) + self.text + "\n"
            result += spaces + "</{0}>\n".format(self.tag)
        else:
            result += "/>\n"

        return result

File: go/xmpp/xmppelement.go
package xmpp

import (
	"strings"
	"fmt"
)

type XmppElement struct {
	Tag        string
	Namespace  string
	Attributes map[string]string
	Children   []*XmppElement
	Text       string
}

func (self *XmppElement) ToXml() (string) {
	return self.ToXmlWithIndent(0)
}

func (self *XmppElement) ToXmlWithIndent(indent int) (string) {
	spaces := strings.Repeat(" ", indent)
	attributes := ""
	for k, v := range self.Attributes {
		attributes += fmt.Sprintf("%s=%s ", k, v)
	}
	result := spaces + fmt.Sprintf("<%s %s", self.Tag, attributes)
	if len(self.Children) > 0 || len(self.Text) > 0 {
		result += ">\n"
		for _, child := range self.Children {
			result += child.ToXmlWithIndent(indent + 4)
		}
		if len(self.Text) > 0 {
			result += strings.Repeat(" ", indent + 4) + self.Text + "\n"
		}
		result += spaces + fmt.Sprintf("</%s>\n", self.Tag)
	} else {
		result += "/>\n"
	}

	return result
}